compiler/repl/evaluate_initScript.sml

```diff
@@ -635,23 +664,60 @@ Proof
     \\ irule v_rel_update
     \\ first_assum (irule_at Any)
     \\ gs [FUN_FMAP_SUBMAP_SUBSET, COUNT_MONO])
+  \\ Cases_on ‘op = FpFromWord’ \\ gs[]
+  >- (
+    gvs[do_app_cases, v_ok_thm]
+    \\ ‘s with <| refs := s.refs; ffi := s.ffi |> = s’ suffices_by gs[]
+    \\ gs[state_component_equality])
+  \\ Cases_on ‘op = FpToWord’ \\ gs[]
+  >- (
+    gvs[do_app_cases, v_ok_thm]
+    \\ ‘s with <| refs := s.refs; ffi := s.ffi |> = s’ suffices_by gs[]
+    \\ gs[state_component_equality])
+  \\ Cases_on ‘op = RealFromFP’ \\ gs[]
+  >- (
+    gvs[do_app_cases, v_ok_thm]
+    \\ ‘s with <| refs := s.refs; ffi := s.ffi |> = s’ suffices_by gs[]
+    \\ gs[state_component_equality])
   \\ Cases_on ‘op’ \\ gs []
 QED
 
 Theorem do_app_ok = SIMP_RULE (srw_ss()) [LET_THM] do_app_ok;
 
+
+val icing_rename_tac =
```

Comment:
Not a fan of custom special-use tactics, esp. in this case when they’re so short.